core/src/main/scala/kafka/metrics/clientmetrics/ClientMetricsCache.scala

```diff
@@ -67,9 +67,10 @@ object  ClientMetricsCache {
    * Launches the asynchronous task to clean the client metric subscriptions that are expired in the cache.
    */
   def runGCIfNeeded(forceGC: Boolean = false): Unit = {
```

Comment:
small nit: to avoid confusion with the actual GC, maybe call this runExpiryIfNeeded?
and change the other gc references to expiry